google/cloud/pubsub/internal/batch_callback_wrapper.h

```diff
@@ -31,22 +32,28 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 class BatchCallbackWrapper : public BatchCallback {
  public:
   using Callback = std::function<void(StreamingPullResponse)>;
+  using MessageCallback = std::function<void(ReceivedMessage)>;
 
   explicit BatchCallbackWrapper(std::shared_ptr<BatchCallback> child,
                                 Callback wrapper)
       : child_(std::move(child)), wrapper_(std::move(wrapper)) {}
+  explicit BatchCallbackWrapper(std::shared_ptr<BatchCallback> child,
+                                MessageCallback wrapper)
+      : child_(std::move(child)), message_wrapper_(std::move(wrapper)) {}
   ~BatchCallbackWrapper() override = default;
 
   void callback(StreamingPullResponse response) override {
     child_->callback(response);
     wrapper_(response);
   }
 
-  void message_callback(MessageCallback::ReceivedMessage m) override {
+  void message_callback(ReceivedMessage m) override {
     child_->message_callback(m);
+    message_wrapper_(m);
   }
 
-  void user_callback(MessageCallback::MessageAndHandler m) override {
+  void user_callback(
+      pubsub_internal::MessageCallback::MessageAndHandler m) override {
```

Comment:
Consider:  The fact that we needed to add qualification here is perhaps an indication that the names are confusing.

Reply:
That is fair. I'm not sure what I would rename it to make it less confusing, so leaving as is for now.